documents/validator.go

```diff
@@ -282,6 +282,17 @@ func ReceivedAnchoredDocumentValidator(
 	}
 }
 
+// RequestDocumentSignatureValidator is a validator group with the following validators
+// SignatureValidator
+// transitionsValidator
+// it should be called when a document is received over the p2p layer before signing
+func RequestDocumentSignatureValidator(idService identity.ServiceDID, senderID identity.DID) ValidatorGroup {
+	return ValidatorGroup{
+		SignatureValidator(idService),
+		transitionValidator(senderID),
```

Comment:
transition validator should be first. This lets us check if collaborator can make the change even before we verify signatures and calculate signing roots